spec/features/ncr/work_orders/edit_spec.rb

```diff
@@ -0,0 +1,331 @@
+      # And we can still submit
+      fill_in 'Vendor', with: 'New ACME'
+      click_on 'Update'
+      expect(current_path).to eq("/proposals/#{ncr_proposal.id}")
+      # Verify it is actually saved
+      work_order.reload
+      expect(work_order.vendor).to eq("New ACME")
+    end
+
+    it "can be edited if approved" do
+      fully_approve(ncr_proposal)
+
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+      expect(current_path).to eq("/ncr/work_orders/#{work_order.id}/edit")
+    end
+
+    it "provides the previous building when editing", :js => true do
+      work_order.update(building_number: "BillDing, street")
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+      click_on "Update"
+      expect(current_path).to eq("/proposals/#{ncr_proposal.id}")
+      expect(work_order.reload.building_number).to eq("BillDing, street")
+    end
+
+    it "allows the user to edit the budget-related fields" do
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+
+      fill_in 'CL number', with: 'CL1234567'
+      fill_in 'Function code', with: 'PG123'
+      fill_in 'Object field / SOC code', with: '789'
+      click_on 'Update'
+
+      work_order.reload
+      expect(work_order.cl_number).to eq('CL1234567')
+      expect(work_order.function_code).to eq('PG123')
+      expect(work_order.soc_code).to eq('789')
+    end
+
+    it "disables the emergency field" do
+      visit "/ncr/work_orders/#{work_order.id}/edit"
+      expect(find_field('emergency', disabled: true)).to be_disabled
+    end
+  end
+
+  it "keeps track of the modification when edited by an approver" do
+    work_order.setup_approvals_and_observers('approver@example.com')
+    approver = ncr_proposal.approvers.last
+    login_as(approver)
+
+    visit "/ncr/work_orders/#{work_order.id}/edit"
+    fill_in 'CL number', with: 'CL1234567'
+    click_on 'Update'
+
+    ncr_proposal.reload
+    update_comments = ncr_proposal.comments.update_comments
+    expect(update_comments.count).to eq(1)
+    # properly attributed
+    update_comment = update_comments.first
+    expect(update_comment.user).to eq(approver)
+    # properly tracked
+    expect(update_comment.comment_text).to include("CL number")
+  end
+
+  it "cannot be edited by someone other than the requester" do
+    stranger = FactoryGirl.create(:user)
+    login_as(stranger)
+
+    visit "/ncr/work_orders/#{work_order.id}/edit"
+    expect(current_path).to eq("/ncr/work_orders/new")
+    expect(page).to have_content("You must be the requester, approver, or observer")
+  end
+
+  describe "post-approval modifications" do
+    def expect_budget_approvals_restarted
+      work_order.reload
+      expect(work_order.status).to eq('pending')
+      approval_statuses = work_order.individual_approvals.pluck(:status)
+      expect(approval_statuses).to eq(%w(
+        approved
+        actionable
+        pending
+      ))
+      # TODO check who gets notified
```

Comment:
@phirefly Who should be notified when the budget approval process gets restarted? What do the notifications need to say?